pkg/client/client.go

```diff
@@ -141,12 +182,21 @@ func (c *populatingClient) Create(ctx context.Context, obj runtime.Object) error
 
 // Update implements Client
 func (c *populatingClient) Update(ctx context.Context, obj runtime.Object) error {
-	meta, client, resource, err := c.metaAndClientFor(obj)
+	meta, client, mapping, resource, err := c.metaAndClientFor(obj)
 	if err != nil {
 		return err
 	}
+	if isNamespaced(mapping) {
+		return client.Put().
+			Namespace(meta.GetNamespace()).
```

Comment:
There is a version of this method called "NamespaceIfScoped" which would simplify this "if" block.

Reply:
Done